addons/stock/models/stock_move_line.py

```diff
@@ -693,7 +693,7 @@ def _free_reservation(self, product_id, location_id, quantity, lot_id=None, pack
 
     def _should_bypass_reservation(self, location):
         self.ensure_one()
-        return location.should_bypass_reservation() or self.product_id.type != 'product'
+        return self.move_id._should_bypass_reservation() or self.product_id.type != 'product'
```

Comment:
You could remove the `self.product_id.type != 'product'` part as you already check it in the `stock.move` `_should_bypass_reservation`, no ?
```suggestion
        return self.move_id._should_bypass_reservation()
```